From 70d2270b0b0cd2c3122e8885aee78a459f953c97 Mon Sep 17 00:00:00 2001
From: Vincent ABRIOU <vincent.abriou@st.com>
Date: Tue, 12 May 2020 13:56:28 +0200
Subject: [PATCH 12/12] dts: av96: add the stm32mp157a-av96-ov5640.dts file

Add the device tree file to support the Avenger96 with the OV5640 D3
mezzanine board.

Signed-off-by: Vincent ABRIOU <vincent.abriou@st.com>
---
 arch/arm/boot/dts/stm32mp157a-av96-ov5640.dts | 103 ++++++++++++++++++
 1 file changed, 103 insertions(+)
 create mode 100644 arch/arm/boot/dts/stm32mp157a-av96-ov5640.dts

diff --git a/arch/arm/boot/dts/stm32mp157a-av96-ov5640.dts b/arch/arm/boot/dts/stm32mp157a-av96-ov5640.dts
new file mode 100644
index 000000000000..3ef922495fed
--- /dev/null
+++ b/arch/arm/boot/dts/stm32mp157a-av96-ov5640.dts
@@ -0,0 +1,103 @@
+// SPDX-License-Identifier: (GPL-2.0+ OR BSD-3-Clause)
+/*
+ * Copyright (C) STMicroelectronics 2020 - All Rights Reserved
+ * Author: Vincent Abriou <vincent.abriou@st.com> for STMicroelectronics.
+ */
+/dts-v1/;
+
+#include "stm32mp157a-av96.dtsi"
+#include <dt-bindings/input/input.h>
+#include <dt-bindings/gpio/gpio.h>
+#include <dt-bindings/soc/stm32-hdp.h>
+
+/ {
+	model = "Arrow Electronics STM32MP157A Avenger96 board (with OV5640 mezzanine board)";
+	compatible = "st,stm32mp157c-ev1", "st,stm32mp157c-ed1", "st,stm32mp157";
+};
+
+&dcmi {
+	status = "okay";
+	pinctrl-names = "default", "sleep";
+	pinctrl-0 = <&dcmi_pins_a>;
+	pinctrl-1 = <&dcmi_sleep_pins_a>;
+
+	port {
+		dcmi_0: endpoint {
+			remote-endpoint = <&mipid02_2>;
+			bus-width = <8>;
+			hsync-active = <0>;
+			vsync-active = <0>;
+			pclk-sample = <0>;
+			pclk-max-frequency = <77000000>;
+		};
+	};
+};
+
+
+&i2c2 {
+	ov5640: camera@3c {
+		compatible = "ovti,ov5640";
+		reg = <0x3c>;
+		pinctrl-names = "default", "sleep";
+		pinctrl-0 = <&rcc_pins_a>;
+		pinctrl-1 = <&rcc_sleep_pins_a>;
+		clocks = <&rcc CK_MCO1>;
+		clock-names = "xclk";
+		assigned-clocks = <&rcc CK_MCO1>;
+		assigned-clock-parents = <&rcc CK_HSE>;
+		assigned-clock-rates = <24000000>;
+		DOVDD-supply = <&v1v8>;
+		reset-gpios = <&gpioa 12 GPIO_ACTIVE_LOW>;/* 31 PA12 TP54 - D3 mezz= GPIOI= TP6 CSI0_RESET (31 low speed) */
+		powerdown-gpios = <&gpiob 5 GPIO_ACTIVE_HIGH>;/* 32 PB5 TP60 - D3 mezz= GPIOJ= TP7 CSI0_PWDN (32 low speed) */
+		rotation = <180>;
+		status = "okay";
+
+		port {
+			ov5640_0: endpoint {
+				remote-endpoint = <&mipid02_0>;
+				clock-lanes = <0>;
+				data-lanes = <1 2>;
+				pclk-max-frequency = <77000000>;
+			};
+		};
+	};
+};
+
+&i2c4 {
+	mipid02: mipid02@14 {
+		compatible = "st,st-mipid02";
+		reg = <0x14>;
+		status = "okay";
+		clocks = <&rcc CK_MCO1>;
+		clock-names = "xclk";
+		assigned-clocks = <&rcc CK_MCO1>;
+		assigned-clock-parents = <&rcc CK_HSE>;
+		assigned-clock-rates = <24000000>;
+		VDDE-supply = <&v1v8>;
+		VDDIN-supply = <&v1v8>;
+		reset-gpios = <&gpioz 0 GPIO_ACTIVE_LOW>;
+		ports {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			port@0 {
+				reg = <0>;
+
+				mipid02_0: endpoint_0 {
+					data-lanes = <1 2>;
+					remote-endpoint = <&ov5640_0>;
+				};
+			};
+			port@2 {
+				reg = <2>;
+
+				mipid02_2: endpoint {
+					bus-width = <8>;
+					hsync-active = <0>;
+					vsync-active = <0>;
+					pclk-sample = <0>;
+					remote-endpoint = <&dcmi_0>;
+				};
+			};
+		};
+	};
+};
-- 
2.17.1

